### DISCUSSION TITLE
Preserve Block Data in Bucket when Destroying a JuiceFS Partition

### DISCUSSION THREAD

**Discussion by michaelmyc** [juicedata/juicefs#4118] [General]
**What would you like to be added**:

The ability to preserve object store bucket when destroying a JuiceFS partition. 

**Why is this needed**:

In case the need to swap metadata engine, I can preferably do the following:
1. dump metadata from old DB
2. load metadata to new DB
3. juicefs destroy (while preserving block data in object store)
4. juicefs format

instead of:
1. dump metadata from old DB
2. mirror block data to new bucket
3. load metadata to new DB
4. juicefs destroy (destroying block data in the process)
5. juicefs format

**Comment by davies** [comment]
You can destroy the database (`drop database xxx` or `flushdb xxx`) directly without using `juicefs destroy`.
